# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../bincimap/x86-64-compile-fix.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- bincimap-1.3.4/src/imapserver.cc.orig	2006-01-04 22:40:26.000000000 +0100
+++ bincimap-1.3.4/src/imapserver.cc	2006-01-04 22:41:42.000000000 +0100
@@ -113,7 +113,7 @@
     BrokerFactory &brokerFactory = BrokerFactory::getInstance();
     Broker *broker = brokerFactory.getBroker(session.getState());
 
-    bincDebug << "IMAPServer::run(), found broker " << (int) broker
+    bincDebug << "IMAPServer::run(), found broker " << long(broker)
 	      << " for state " << session.getState() << endl;
 
     bool skipToNextRequest = false;
--- bincimap-1.3.4/src/convert.h.orig	2006-01-04 22:42:12.000000000 +0100
+++ bincimap-1.3.4/src/convert.h	2006-01-04 22:47:58.000000000 +0100
@@ -59,6 +59,14 @@
   }
 
   //----------------------------------------------------------------------
+  inline std::string toString(long i_in)
+  {
+    char longbuf[40];
+    snprintf(longbuf, sizeof(longbuf), "%ld", i_in);
+    return std::string(longbuf);
+  }
+
+  //----------------------------------------------------------------------
   inline std::string toString(unsigned long i_in)
   {
     char longbuf[40];
@@ -293,6 +301,8 @@
     BincStream &operator << (const std::string &t);
     BincStream &operator << (unsigned int t);
     BincStream &operator << (int t);
+    BincStream &operator << (unsigned long t);
+    BincStream &operator << (long t);
     BincStream &operator << (char t);
 
     //--
--- bincimap-1.3.4/src/convert.cc.orig	2006-01-04 22:44:24.000000000 +0100
+++ bincimap-1.3.4/src/convert.cc	2006-01-04 22:48:34.000000000 +0100
@@ -124,6 +124,20 @@
 }
 
 //------------------------------------------------------------------------
+BincStream &BincStream::operator << (long t)
+{
+  nstr += toString(t);
+  return *this;
+}
+
+//------------------------------------------------------------------------
+BincStream &BincStream::operator << (unsigned long t)
+{
+  nstr += toString(t);
+  return *this;
+}
+
+//------------------------------------------------------------------------
 BincStream &BincStream::operator << (char t)
 {
   nstr += t;
